compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1969,16 +2056,26 @@ impl<'a> Resolver<'a> {
                     String::from("add `extern crate alloc` to use the `alloc` crate"),
                     Applicability::MaybeIncorrect,
                 ))
+            } else if let Some(sugg) =
+                self.find_similarly_named_module_or_crate(ident.name, &parent_scope.module)
+            {
+                Some((
+                    vec![(ident.span, sugg.to_string())],
+                    String::from("there is a crate or module with a similar name"),
```

Comment:
I don't see a test that covers this case. Can you write one?

Reply:
I think the following test covers that case, doesn't it?
https://github.com/rust-lang/rust/blob/master/src/test/ui/suggestions/crate-or-module-typo.rs